MetaMask/metamask-mobile
# Fix: Android Shasum Validation

## **Description**

Before :

this change we needed to disable shasum validation on snaps. The reason for this was due to the fact that my custom RnTar native module was reading the parsed snap code by line. This lead to small differences in the output file that differed from the npm source code. The result was that the computed shasum would be different that the one present in the manifest and woulld fail to install.

Now:
- Now we read the source code file by file which maintains the integrity and results in a matching shasum. 
- I also converted this module to run on a separate thread so we don't block the main thread which can lead to perceived performance drop.

## **Related issues**

Fixes: https://github.com/MetaMask/mobile-planning/issues/952

## **Manual testing steps**

1. You need android studio for this change
2. pull branch
3. run `yarn setup`
4. run `start:android`
5. open the app in android studio
6. alternativily you can run in android studio
7. create wallet
8. navigate to https://metamask.github.io/snaps/test-snaps/latest
9. install a snap
10. the snap should prompt the user with a connection request, permission request and success screen

## **Screenshots/Recordings**
<img width="1061" alt="Screenshot 2023-11-10 at 10 22 33 AM" src="https://github.com/MetaMask/metamask-mobile/assets/22918444/3c791adc-e100-4f4c-a58c-05055715107d">


https://github.com/MetaMask/metamask-mobile/assets/22918444/248421ce-84f7-4f6d-aa88-c8133be23812


## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
